File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use App\Repository\EvenementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use phpDocumentor\Reflection\DocBlock\Tags\PropertyRead;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups ;

#[ORM\Entity(repositoryClass: EvenementRepository::class)]
class Evenement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups("post:read")]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\Length(
        min :2 ,
        max :15,
        minMessage: 'le nom doit etre supperieur à {{ limit }} caracteres',
        maxMessage: 'le nom ne doit pas dépasser {{ limit }} caracteres',)]
    #[Assert\NotNull]
    #[Groups("post:read")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Groups("post:read")]
    private ?string $lieu = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\GreaterThan("today")]
    #[Groups("post:read")]
    private ?\DateTimeInterface $date = null;

    #[ORM\Column(length: 255)]
    #[Assert\Length(
        min :5 ,
        max :50,
        minMessage: 'la description doit etre supperieur à {{ limit }} caracteres',
        maxMessage: 'la description ne doit pas dépasser {{ limit }} caracteres',)]
    #[Assert\NotNull]
    #[Assert\NotBlank( message:"Votre description ne doit pas etre vide")]
    #[Groups("post:read")]
    private ?string $description = null;


    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\GreaterThan("today")]
    #[Groups("post:read")]
    private ?\DateTimeInterface $datefin = null;


    #[ORM\OneToMany(mappedBy: 'evenement', targetEntity: Participation::class)]
    #[Groups("post:read")]
    private Collection $participations;

    #[ORM\Column(length: 255)]
    #[Groups("post:read")]
    private ?string $image = null;

    #[ORM\Column]
    #[Groups("post:read")]
    private ?int $nbrPersonnes = null;







    public function __construct()
    {
        $this->participations = new ArrayCollection();

    }




    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getLieu(): ?string
    {
        return $this->lieu;
    }

    public function setLieu(string $lieu): self
    {
        $this->lieu = $lieu;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getDatefin(): ?\DateTimeInterface
    {
        return $this->datefin;
    }

    public function setDatefin(\DateTimeInterface $datefin): self
    {
        $this->datefin = $datefin;

        return $this;
    }
    public function __toString()
    {
        return(string)$this->getDate();
    }

    /**
     * @return Collection<int, Participation>
     */
    public function getParticipations(): Collection
    {
        return $this->participations;
    }

    public function addParticipation(Participation $participation): self
    {
        if (!$this->participations->contains($participation)) {
            $this->participations->add($participation);
            $participation->setEvenement($this);
        }

        return $this;
    }

    public function removeParticipation(Participation $participation): self
    {
        if ($this->participations->removeElement($participation)) {
            // set the owning side to null (unless already changed)
            if ($participation->getEvenement() === $this) {
                $participation->setEvenement(null);
            }
        }

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getNbrPersonnes(): ?int
    {
        return $this->nbrPersonnes;
    }

    public function setNbrPersonnes(int $nbrPersonnes): self
    {
        $this->nbrPersonnes = $nbrPersonnes;

        return $this;
    }





}

File: src/users/entity/Evenement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package users.entity;

import java.sql.Date;

/**
 *
 * @author joujo
 */
public class Evenement {
    
      private int id;
    String nom;
    private String lieu;
    private Date date;
    private String description;

    private Date datefin;
    private int nbr_personnes;
    

    public Evenement() {
    }

    public Evenement(int id) {
        this.id = id;
    }

    public Evenement(int id, String nom, String lieu, Date date, String description, Date datefin, int nbr_personnes) {
        this.id = id;
        this.nom = nom;
        this.lieu = lieu;
        this.date = date;
        this.description = description;
        this.datefin = datefin;
        this.nbr_personnes = nbr_personnes;
    }

    public Evenement(String nom, String lieu, Date date, String description,Date datefin, int nbr_personnes) {
        this.nom = nom;
        this.lieu = lieu;
        this.date = date;
        this.description = description;
        this.datefin = datefin;
        this.nbr_personnes = nbr_personnes;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

 

    public Date getDatefin() {
        return datefin;
    }

    public void setDatefin(Date datefin) {
        this.datefin = datefin;
    }

    public int getNbr_personnes() {
        return nbr_personnes;
    }

    public void setNbr_personnes(int nbr_personnes) {
        this.nbr_personnes = nbr_personnes;
    }

    @Override
    public int hashCode() {
        int hash = 5;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Evenement other = (Evenement) obj;
        if (this.id != other.id) {
            return false;
        }
        return true;
    }
    

    @Override
    public String toString() {
        return "Evenement{" + "id=" + id + ", nom=" + nom + ", lieu=" + lieu + ", date=" + date + ", description=" + description +", datefin=" + datefin + ", nbr_personnes=" + nbr_personnes + '}';
    }
    

    
}
